src/main/java/backend/Logic.java

```diff
@@ -271,6 +271,28 @@ public Model getRepo(String repoId) {
     }
 
     /**
+     * Replaces existing milestone with the newMilestone in the issue object, the UI, and the server, in that order.
+     * Server update is done after the local update to reduce the lag between the user action and the UI response
+     *
+     * @param issue The issue object whose milestone is to be replaced.
+     * @param newMilestone The new milestone to be assigned to the issue
+     * @return true if milestone replacement on GitHub was a success, false otherwise.
+     */
+    public CompletableFuture<Boolean> replaceIssueMilestone(TurboIssue issue, Integer newMilestone) {
+        Optional<Integer> originalMilestone = issue.getMilestone();
+
+        logger.info("Changing milestone for " + issue + " on UI");
+        CompletableFuture<Optional<TurboIssue>> localMilestoneReplaceFuture =
```

Comment:
I don't think the result should be optional. It reflects the error condition of not being able to find an issue in the model, which is very likely to be programmer error. (Being optional means that the value may reasonably be absent, and that the programmer should remember to handle that possibility.)

This is probably better handled by throwing an exception, then using `exceptionally` to log and return `false` to the caller.